File: findPairs.py
class Solution:
    def findPairs(self, nums: List[int], k: int) -> int:
        hashmap = dict()
        count = 0
        # creating hte frequency dict
        for i in range(len(nums)):
            if nums[i] not in hashmap:
                hashmap[nums[i]] = 1
            else:
                temp = hashmap[nums[i]]
                hashmap[nums[i]] = temp + 1

        # iterate over the hashmap for )elemt - target being presnt in the hashmap
        for x, y in hashmap.items():
            if (x - k) in hashmap:
                if k != 0:
                    count += 1
                else:
                    if y >= 2:
                        count += 1

        return count

File: Find_Pairs.java
class Solution {
    public int findPairs(int[] nums, int k) {
        HashMap <Integer, Integer> hm = new HashMap<>();
        
        for(int i : nums){
            hm.put(i , hm.getOrDefault(i , 0) + 1);
        }
        int count = 0;
        for(int i : hm.keySet()){
            if((k==0 && hm.get(i) > 1) || (k> 0 && hm.containsKey(i+k) ) ) {
                count++;
        
        if(k==0)
            hm.put(i , hm.get(i) - 2);
            
        else{
            hm.put( i , hm.get(i) - 1);
         
            hm.put(i + k , hm.get(i + k) - 1);
            
        }
            }
        }
        return count;
    }
}







